features/environment.py

```diff
@@ -498,22 +499,47 @@ def after_scenario(context, scenario):
                 logging.warning("Failed to collect coverage")
 
 
-def _get_apparmor_logs_from_host():
-    # get apparmor DENIED messages from the host
-    with open("/var/log/syslog", "r") as syslog_fd:
-        syslog_messages = syslog_fd.readlines()
-        apparmor_denied = [
-            msg.strip()
-            for msg in syslog_messages
-            if ("DENIED" in msg and "ubuntu_pro_" in msg)
-        ]
-    return apparmor_denied
+def _get_relevant_apparmor_logs(context):
+    if hasattr(context, "machines") and SUT in context.machines:
+        sut = context.machines[SUT]
+        if sut.cloud == "lxd-container":
+            # get apparmor DENIED messages from the host
+            with open("/var/log/syslog", "r") as syslog_fd:
+                syslog_messages = syslog_fd.readlines()
```

Comment:
In the case of a lxd container, yes, we need to check for apparmor errors on the host's /var/log/syslog.